src/libraries/Microsoft.PowerFx.Interpreter/Functions/FormattingInfo.cs

```diff
@@ -6,10 +6,23 @@
 
 namespace Microsoft.PowerFx.Functions
 {
+    [ThreadSafeImmutable]
     internal class FormattingInfo
     {
-        public CultureInfo CultureInfo { get; set; }
+        public readonly CultureInfo CultureInfo;
+        public readonly TimeZoneInfo TimeZoneInfo;
 
-        public TimeZoneInfo TimeZoneInfo { get; set; }
+        public FormattingInfo()
+        {
+            // $$$ can't use current culture
+            CultureInfo = CultureInfo.CurrentCulture;
```

Comment:
can we avoid adding the reference to CurrentCulture?

Reply:
move this to helper class.
they can pass in null, they will get back null for culture and timezone if do so.